prepare 2.9.0.4

gerhardol opened the pull request:
Still get a lint for an icon


C:\dev\gc\runnerup\wear\src\main\java\org\runnerup\service\ListenerService.java:271: Warning: The static icon for an ongoing activity should be white with a transparent background [ActivityIconColor]
        .setStaticIcon(R.drawable.ic_launcher)
                       ~~~~~~~~~~~~~~~~~~~~~~

   Explanation for issues of type "ActivityIconColor":
   The resources passed to setAnimatedIcon and setStaticIcon should be white
   with a transparent background, preferably a VectorDrawable or
   AnimatedVectorDrawable.

   https://developer.android.com/training/wearables/ongoing-activity#best-practices

gerhardol commented:
@roberi 
Thanks, that could work (have not tested).
I added it to the PR,. (The vector image is only the lines, so only foreground lines.)

jonasoreland commented:
fwiw, tested...and couldn't see the icon anyway,
not sure if i'm running the incorrect version of emulator...or what

if the lint warning (that I haven't found!) is gone, then I say go for it!

roberi commented:
> fwiw, tested...and couldn't see the icon anyway,
> not sure if i'm running the incorrect version of emulator...or what

Working for me on Wear 3 (emulator API 30) 
![image](https://github.com/user-attachments/assets/e718b64d-fe2b-4ec9-bb1c-4d2a09d29c63)

jonasoreland commented:
nice!

roberi commented:
@jonasoreland, I noticed two more issues while testing the ongoing notification. They probably existed before the latest changes. Fix them here, or in a separate PR?

![image](https://github.com/user-attachments/assets/795fecfa-4fc5-4bcb-817b-ab65636032f6)

Nothing but an output in the phone's logcat happens when pressing "Start GPS" on the watch (Wear 3):
onMessageReceived: MessageEventParcelable[16560,/org.runnerup/workout/start, size=0]

The time card needs some fine-tuning.

jonasoreland commented:
1) It is a "know problem" that Start GPS is in fact not implemented...
I suggest to skip that

2) If you can fix the time card (make sure 24h format) ? that would be great...I think here is fine...

gerhardol commented:
@roberi 
The Play feedback has commented on various clipped images. If there is an easy fix please push to this branch.
I plan to submit to Play review late this evening when I get home.

roberi commented:
Adding this line to the Card1TextLarge style is a good enough fix for now.

`<item name="android:textAlignment">center</item>`

![image](https://github.com/user-attachments/assets/1d9e52ce-d0ed-4ac0-bb43-59f167b8f70d)

gerhardol commented:
I updated the time entry as well as the other cards, centered the text
The lint was not complete gone, updated again
Suppressed the permission lint that Jonas had tested. Should probably be fixed anyway, at some point

gerhardol commented:
submitted update to Play after merging #1236 too
